packages/flutter/lib/src/cupertino/thumb_painter.dart

```diff
@@ -39,22 +26,16 @@ const List<BoxShadow> _kSliderBoxShadows = <BoxShadow> [
   ),
 ];
 
-/// Paints an iOS-style slider thumb or switch thumb.
+/// Paints an iOS-style slider thumb.
 ///
-/// Used by [CupertinoSwitch] and [CupertinoSlider].
+/// Used by [CupertinoSlider].
```

Comment:
Why not use CupertinoThumbPainter? Is it out of date?

Reply:
`CupertinoThumbPainter` doesn't paint an image/icon on the thumb. I see some duplicated code, especially in the  `_paintCupertinoThumbShadowAndBorder` method as compared to `CupertinoThumbPainter`'s paint method. I thought it would be a challenge to refactor `CupertinoThumbPainter` to use this new implementation, so I just decided to not use it for now.